File: 28.find-the-index-of-the-first-occurrence-in-a-string.py
#
# @lc app=leetcode id=28 lang=python3
#
# [28] Find the Index of the First Occurrence in a String
#

# @lc code=start
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        lastIndex = len(haystack)-len(needle)
        if lastIndex<0:
            return -1
        
        for i in range(len(haystack)):
            if haystack[i:i+len(needle)]==needle:
                return i
        return -1
        
# @lc code=end

File: 28.find-the-index-of-the-first-occurrence-in-a-string.go
/*
 * @lc app=leetcode id=28 lang=golang
 *
 * [28] Find the Index of the First Occurrence in a String
 */

// @lc code=start
package main

/*
n = len(needle)
m = len(haystack)
for i = 0;i < m-1;i++

	substr = haystack[i:i+n] // if out of range, ignore
	if substr == needle:
	  return i

return -1
*/
func strStr(haystack string, needle string) int {
	n := len(needle)
	m := len(haystack)
	for i, _ := range haystack {
		if i+n > m {
			continue
		}
		substr := haystack[i : i+n]
		if substr == needle {
			return i
		}
	}
	return -1
}

// @lc code=end
